cou/utils/juju_utils.py

```diff
@@ -326,12 +327,20 @@ async def run_on_unit(
         :returns: action.data['results'] {'Code': '', 'Stderr': '', 'Stdout': ''}
         :rtype: dict[str, str]
         :raises UnitNotFound: When a valid unit cannot be found.
-        :raises ActionFailed: When the application status is in error (it's not 'completed').
+        :raises CommandRunFailed: When a command fails to run.
         """
+        logger.debug("Running '%s' on '%s'", command, unit_name)
+
         unit = await self._get_unit(unit_name)
         action = await unit.run(command, timeout=timeout)
         results = action.data.get("results")
-        return _normalize_action_results(results)
+        normalize_results = _normalize_action_results(results)
+
+        if str(normalize_results["Code"]) != "0":
```

Comment:
Should we use same approach as we are doing in `run_action`? I mean to use `raise_on_failure`.

Reply:
I don't think so. The point of having `raise_on_failure` for actions is because we are retrying for getting action result. But for run on command, getting the result is suppose to be static and no need for retrying.

Reply:
But what if I want to run command and I expecting that it could failed?

Reply:
Opened #148 to track

Reply:
Thanks